dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -1039,4 +1033,12 @@ private List<Tool> updateTools(final Iterable<Tool> apiToolList, final List<Tool
 
         return dbToolList;
     }
+
+    public LanguageHandlerInterface.DockerSpecifier getSpecifierFromTagName(String tagName) {
+        if ("latest".equals(tagName)) {
+            return LanguageHandlerInterface.DockerSpecifier.LATEST;
+        } else {
+            return LanguageHandlerInterface.DockerSpecifier.TAG;
```

Comment:
I'm not entirely understanding the flow, and it was already like this up on line 730, but is it possible that the `tagName` could be any of the other `DockerSpecifier` enums?

Reply:
This function assumes that it's only used by tools, which I believe can only have the latest tag or a non-latest tag. I don't think they can have no tag or a digest, is that right?

Reply:
I think so, assuming this code path doesn't include apptools

Reply:
> This PR also adds multi-arch Quay support for workflows (regular and GitHub apps) and GitHub app tools. I made several QuayImageRegistry methods public and refactored them a bit so that LanguageHandlerInterface can re-use them when processing images for workflows and GitHub app tools to reduce duplication.

Based on this and Denis' comment, is it possible the method is now being called to process an app tool or workflow?

Reply:
No, apptools and workflows get their images processed in `LanguageHandlerInterface` [here](https://github.com/dockstore/dockstore/blob/859f7eef233f877df71b740514dd21c2d2592150/dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java#L578) and the images themselves come from the tool json table